test/util/mock-server.js

```diff
@@ -0,0 +1,63 @@
+import express from 'express';
+import bodyParser from 'body-parser';
+import { assert } from 'chai';
+
+const NOT_LISTENING = 0;
+const LISTENING = 1;
+const AWAITING_VALIDATION = 2;
+
+/**
+ * See: docs/Mock_Server.md
+ */
+
+class MockServer {
+
+    constructor() {
+        this.status = NOT_LISTENING;
+        this.server = null;
+        this.request = null;
+    }
+
+    listen(config) {
+        if (this.status === LISTENING) {
+            throw new Error('Mock server is already listening for a request.');
+        }
+        if (this.status === AWAITING_VALIDATION) {
+            throw new Error('Mock server is waiting for the request to be validated.');
+        }
+        this.status = LISTENING;
+        return new Promise(resolve => {
+            let app = express();
+            app.use(bodyParser.json());
+            app[config.method](config.endpoint, (req, res) => {
+                if (this.request !== null) { // TODO: test what happens with this error
```

Comment:
Yep, I forgot to handle this `TODO`. It was a note-to-self, not something I meant to include in this PR.